module/VuFind/src/VuFind/Export/Zotero/ZoteroService.php

```diff
@@ -0,0 +1,395 @@
+            );
+        } catch (CredentialsException $e) {
+            $this->logError('Could not retrieve token credentials from Zotero: ' . (string)$e);
+            throw new ZoteroException('An error has occurred');
+        }
+        assert($credentials instanceof ZoteroTokenCredentials);
+        unset($this->sessionContainer->tempCredentials);
+        // Create access token:
+        $accessToken = $this->accessTokenService->getByIdAndType((string)$user->getId(), self::ACCESS_TOKEN_TYPE);
+        $accessToken
+            ->setUser($user)
+            ->setData(
+                json_encode(
+                    [
+                        'zoteroUserId' => $credentials->getUserId(),
+                        'zoteroApiKey' => $credentials->getSecret(),
+                    ]
+                )
+            );
+        $this->accessTokenService->persistEntity($accessToken);
+        // Perform the actual export:
+        return $this->exportToZotero($accessToken);
+    }
+
+    /**
+     * Get temporary credentials from Zotero and return an authorization URL.
+     *
+     * @return string
+     *
+     * @throws ZoteroException
+     */
+    protected function getZoteroAuthUrlWithTempCredentials(): string
+    {
+        // Get request token from Zotero:
+        try {
+            $this->sessionContainer->tempCredentials = $this->zoteroOAuth->getTemporaryCredentials();
+        } catch (CredentialsException $e) {
+            $this->logError('Could not retrieve temporary credentials from Zotero: ' . (string)$e);
+            throw new ZoteroException('Could not retrieve temporary credentials from Zotero');
+        }
+
+        // Redirect to Zotero for authentication:
+        return $this->zoteroOAuth->getAuthorizationUrl($this->sessionContainer->tempCredentials);
+    }
+
+    /**
+     * Export record(s) to Zotero using its v3 API.
+     *
+     * @param AccessTokenEntityInterface $accessToken Access token with Zotero credentials
+     *
+     * @return int Exported record count
+     *
+     * @throws ZoteroException
+     */
+    protected function exportToZotero(AccessTokenEntityInterface $accessToken): int
+    {
+        if (!($callbackUrl = $this->sessionContainer->exportCallbackUrl)) {
+            $this->logError('Export callback URL not available in session');
+            throw new ZoteroException('Export callback URL not available in session');
+        }
+
+        try {
+            $vufindResponse = $this->httpService->createClient($callbackUrl)->request('GET', $callbackUrl);
+        } catch (\Exception $e) {
+            $this->logError("GET request for '$callbackUrl' failed: " . (string)$e);
+            throw new ZoteroException('Export request failed');
+        }
+
+        if ($vufindResponse->getStatusCode() !== 200) {
+            $this->logError(
+                "GET request for '$callbackUrl' failed: "
+                . $vufindResponse->getStatusCode() . ': ' . $vufindResponse-> getReasonPhrase()
```

Comment:
Random extra space here:

```suggestion
                . $vufindResponse->getStatusCode() . ': ' . $vufindResponse->getReasonPhrase()
```